src/molecules/Amount/Amount.native.js

```diff
@@ -0,0 +1,169 @@
+import automation from '../../_helpers/automation-attributes';
+import Flex from '../../atoms/Flex';
+import Text from '../../atoms/Text';
+import Heading from '../../atoms/Heading';
+
+const styles = {
+  text: {
+    color({ size }) {
+      switch (size) {
+        case 'medium':
+          return 'shade.940';
+        case 'large':
+          return 'shade.940';
+        case 'xlarge':
+          return 'shade.940';
+        case 'xxlarge':
+          return 'shade.940';
+        case 'xxxlarge':
+          return 'shade.950';
+        default:
+          return 'shade.940';
+      }
+    },
+    size({ size }) {
+      switch (size) {
+        case 'medium':
+          return 'xxsmall';
+        case 'large':
+          return 'xsmall';
+        case 'xlarge':
+          return 'medium';
+        case 'xxlarge':
+          return 'medium';
+        case 'xxxlarge':
+          return 'large';
+        default:
+          return 'xxsmall';
+      }
+    },
+    weight({ size }) {
+      switch (size) {
+        case 'medium':
+          return 'regular';
+        case 'large':
+          return 'bold';
+        case 'xlarge':
+          return 'bold';
+        case 'xxlarge':
+          return 'bold';
+        case 'xxxlarge':
+          return 'bold';
+        default:
+          return 'bold';
+      }
+    },
+    lineHeight({ size }) {
+      switch (size) {
+        case 'medium':
+          return 'medium';
+        case 'large':
+          return 'medium';
+        case 'xlarge':
+          return 'xlarge';
+        case 'xxlarge':
+          return 'xxlarge';
+        case 'xxxlarge':
+          return 'xxlarge';
+        default:
+          return 'xxlarge';
+      }
+    },
+  },
+  baseText: {
+    alignSelf({ align }) {
+      switch (align) {
+        case 'left':
+          return 'flex-start';
+        case 'center':
+          return 'center';
+        case 'right':
+          return 'flex-end';
+        default:
+          return 'flex-start';
+      }
+    },
+  },
+};
+
+const BaseText = styled(NativeText)`
+  align-self: ${styles.baseText.alignSelf};
+`;
+
+const getFractionalDisplay = (fractionalPart = '') => {
+  switch (fractionalPart.length) {
+    case 0:
+      return '00';
+    case 1:
+      return `${fractionalPart}0`;
+    case 2:
+      return fractionalPart;
+    default:
+      return fractionalPart.slice(0, 2);
+  }
+};
+
+const getLocaleString = (numberString) => {
+  if (Platform.OS === 'android') {
+    // only android needs this polyfill
+    require('intl');
+    require('intl/locale-data/jsonp/en-IN');
+    return new Intl.NumberFormat('en-IN').format(numberString);
+  } else {
+    return parseInt(numberString, 10).toLocaleString('en-IN');
+  }
+};
+
+const Amount = ({ size, align, testID, children }) => {
+  if (isNaN(children)) {
+    throw new Error(`Expected children to be number \n(Eg. "1234", "12.34")`);
+  }
+
+  const currency = '₹';
+
+  const [integerPart, fractionPart] = children.split('.');
```

Comment:
Wouldn't this fail like the `fractionPart` will be `undefined`? The case where the user will enter `123`.
You can do it this way
```js
const [integerPart, fractionPart] = String(parseFloat(amount).toFixed(2)).split('.')
```

Reply:
It won't fail because ```const getFractionalDisplay = (fractionalPart = '') => {```

But yeah, I can try the ```toFixed``` thing